File: 1941-check-if-all-characters-have-equal-number-of-occurrences/1941-check-if-all-characters-have-equal-number-of-occurrences.py
class Solution:
    def areOccurrencesEqual(self, s: str) -> bool:
        hash_map = {}
        for i in s:
            if i in hash_map:
                hash_map[i] += 1
            else:
                hash_map[i] = 1
        
        previous = None
        for key in hash_map:
            if previous is None:
                previous = hash_map[key]
                continue
            if hash_map[key] != previous:
                return False
        
        return True

File: 1941-check-if-all-characters-have-equal-number-of-occurrences/1941-check-if-all-characters-have-equal-number-of-occurrences.java
class Solution {
    public boolean areOccurrencesEqual(String s) {
        Map<Character, Integer> map = new HashMap<>();
        for (char character: s.toCharArray()) {
            map.put(character, map.getOrDefault(character, 0) + 1);
        }
        int frequency = 0;
        for (Map.Entry<Character, Integer> mapEntry: map.entrySet()) {
            if (frequency == 0) {
                frequency = mapEntry.getValue();
                continue;
            }
            if (mapEntry.getValue() != frequency) {
                return false;
            }
        }
        return true;
    }
}